src/sagas/roles.js

```diff
@@ -0,0 +1,334 @@
+// import dayjs from 'dayjs';
+// import nanoid from 'nanoid';
+import {
+  all,
+  call,
+  put,
+  takeLatest
+  // delay,
+  // take,
+  // takeEvery,
+  // select
+} from 'redux-saga/effects';
+
+import request from 'utils/request';
+import { actions, types } from 'reducers/roles';
+import { actions as appActions } from 'reducers/application';
+
+// snake_cased variables here come from RFC 6749
+/* eslint-disable camelcase */
+function* requestRolesWorker() {
+  try {
+    const endpoint = {
+      url: '/roles',
+      method: 'GET'
+    };
+    const result = yield call(request.execute, { endpoint });
+
+    // update roles in state or throw an error
+    if (result.success) {
+      const {
+        response: { data }
+      } = result;
+
+      yield put(actions.requestRolesSuccess(data));
+    } else if (result.error) {
+      throw result.error;
+    } else {
+      throw new Error('Failed to get roles!');
+    }
+  } catch (error) {
+    const { message } = error;
+
+    yield put(actions.requestRolesFailure(error));
+    yield put(
+      appActions.popToast({
+        title: 'Error',
+        icon: 'times-circle',
+        message
+      })
+    );
+  }
+}
+
+function* createRoleWorker({ name }) {
```

Comment:
There's a pretty big code coverage gap in this file.

Reply:
Added coverage for this file